micrometer-core/src/main/java/io/micrometer/core/instrument/binder/httpcomponents/MicrometerHttpRequestExecutor.java

```diff
@@ -195,12 +282,40 @@ public Builder exportTagsForRoute(boolean exportTagsForRoute) {
             return this;
         }
 
+        /**
+         * Configure an observation registry to instrument using the {@link Observation}
+         * API instead of directly with a {@link Timer}.
+         * @param observationRegistry registry with which to instrument
+         * @return This builder instance.
+         */
+        public Builder observationRegistry(ObservationRegistry observationRegistry) {
+            this.observationRegistry = observationRegistry;
+            return this;
+        }
+
+        /**
+         * Provide a custom convention to override the default convention used when
+         * instrumenting with the {@link Observation} API. This only takes effect when a
+         * {@link #observationRegistry(ObservationRegistry)} is configured.
+         * @param convention semantic convention to use
+         * @return This builder instance.
+         * @see #observationRegistry(ObservationRegistry)
+         */
+        public Builder observationConvention(ApacheHttpClientObservationConvention convention) {
+            this.observationConvention = convention;
+            return this;
+        }
+
         /**
          * @return Creates an instance of {@link MicrometerHttpRequestExecutor} with all
          * the configured properties.
          */
         public MicrometerHttpRequestExecutor build() {
-            return new MicrometerHttpRequestExecutor(waitForContinue, registry, uriMapper, tags, exportTagsForRoute);
+            if (observationConvention == null) {
+                observationConvention = new DefaultApacheHttpClientObservationConvention(uriMapper, exportTagsForRoute);
```

Comment:
I guess it will be needed to move `uriMapper` and `exportTagsForRoute` to the context. As it is, a user-provided convention would not have access to these.